pkg/common/common.go

```diff
@@ -198,3 +201,18 @@ func CCOWorkflow() bool {
 	}
 	return false
 }
+
+// StripDefaultPorts removes port 80 from HTTP URLs and 443 from HTTPS URLs.
+// Defer to the actual AWS SDK implementation to match its behavior exactly.
+func StripDefaultPorts(fromUrl string) (string, error) {
```

Comment:
here I ran a test focusing on IBM and was not successful

```go
func TestStripDefaultPorts(t *testing.T) {
	base := "https://s3.region.cloud-object-storage.appdomain.cloud/bucket-name"
	want := "https://s3.region.cloud-object-storage.appdomain.cloud/bucket-name"
	t.Run("original URL is returned", func(t *testing.T) {
		got, err := StripDefaultPorts(base)
		if err != nil {
			t.Errorf("An error occurred: %v", err)
		}
		if !reflect.DeepEqual(got, want) {
			t.Errorf("StripDefaultPorts() = %v, want %v", got, want)
		}
	})
}
```

Reply:
Add a test case and fix?

Reply:
I think this is a new test case, right Mateus? Thanks for adding this. In this test, StripDefaultPorts correctly extracts the host name from the URL, but then overwrites it with a blank string. That blank string comes from an HTTP request field that is not blank when running against a real AWS cluster. I will add a fix that should get this test working.

Reply:
This should fix it: https://github.com/openshift/oadp-operator/pull/1177